### DISCUSSION TITLE
Useless UniqueConstraint in docs example?

### DISCUSSION THREAD

**Discussion by lafrech** [sqlalchemy/sqlalchemy#10434] [Usage Questions]
I have a question regarding an example mapping found [in the docs](https://github.com/sqlalchemy/sqlalchemy/blob/6d6e3df8ce0d455eac00ddc8d9618897ae799fe4/doc/build/orm/relationship_persistence.rst):

```py
class Entry(Base):
    __tablename__ = "entry"
    entry_id = mapped_column(Integer, primary_key=True)
    widget_id = mapped_column(Integer, ForeignKey("widget.widget_id"))
    name = mapped_column(String(50))
    __table_args__ = (UniqueConstraint("entry_id", "widget_id"),)
```

I don't understand the point of the additional `UniqueConstraint`. Shouldn't `entry_id` be unique by itself as a primary key?

**Comment by lafrech** [comment]
OK, after playing with this a bit, I realized that without this constraint definition, the definition below fails:

```py
    __table_args__ = (
        ForeignKeyConstraint(
            ["widget_id", "favorite_entry_id"],
            ["entry.widget_id", "entry.entry_id"],  # <= This must be an existing unique constraint
            name="fk_favorite_entry",
        ),
    )
```

It seems `UniqueConstraint("entry_id", "widget_id")` must be declared explicitly for this to work.

**Reply by CaselIT** [reply]
I guess some db require an unique contraint for foreign keys target.
Feel free to open a PR to add a comment in the code snippet indicating that the unique contraint is required by sone db to allow the foreign key
